### PR TITLE
Surfaceartifacts

### PR DESCRIPTION
**Pull request by stopthatastronaut**
Picks up and surfaces Octopus Logs from the vagrant box's disk, allowing us to have a spelunk in there without logging on to the box

Logs surface in Team City under the "Scenario_xx_Install" block, in a sub-block entitled "Logfiles".

Example build log:

image

### CODE REVIEW

File: OctopusDSC/DSCResources/cOctopusServer/cOctopusServer.psm1 [hunk 1]
```diff
@@ -5,7 +5,7 @@ $script:instancecontext = ''  # a global to hold the name of the current instanc
 # dot-source the helper file (cannot load as a module due to scope considerations)
 . (Join-Path -Path (Split-Path (Split-Path $PSScriptRoot -Parent) -Parent) -ChildPath 'OctopusDSCHelpers.ps1')
 
-function Resolve-Error
+function Resolve-ODSCError
```
Comment (matt-richardson) [inline OctopusDSC/DSCResources/cOctopusServer/cOctopusServer.psm1]: `Resolve-ODSCError` isn't very clear... How about `Resolve-OctopusDSCError`?

File: build-hyperv.ps1 [hunk 2]
```diff
@@ -77,7 +79,7 @@ if ($result.FailedCount -gt 0) {
 }
 
 Write-Output "Running 'vagrant up --provider hyperv'"
-vagrant up --provider hyperv --no-destroy-on-error | Tee-Object -FilePath vagrant.log  #   --debug
+vagrant up --provider hyperv --no-destroy-on-error --debug | Tee-Object -FilePath vagrant.log  #   --debug
```
Comment (matt-richardson) [inline build-hyperv.ps1]: Dont want the `--debug` committed

File: vagrantfile [hunk 3]
```diff
@@ -77,7 +79,7 @@ Vagrant.configure(VAGRANT_FILE_API_VERSION) do |config|
     config.vm.provider "hyperv" do |v, override|
       override.vm.box = "gusztavvargadr/w16s-sql17d"   # temporary box url. We'll need to build one eventually
       config.vm.network "public_network", bridge: "External Connection"
-      override.vm.synced_folder ".", "/vagrant", type: "smb", smb_host: "192.168.1.3" # this is required or it guesses the wrong IP
+      # override.vm.synced_folder ".", "/vagrant", type: "smb", smb_host: "192.168.1.3" # this is required or it guesses the wrong IP
```
Comment (matt-richardson) [inline vagrantfile]: Do we want to remove this? if we do, should we kill this line?

Comment (stopthatastronaut) [reply vagrantfile]: Seems to work OK without it, I'll kill it.

**Pull request merged**